src/utils/getContractsConsumption.ts

```diff
@@ -20,6 +20,6 @@ export default function getContractsConsumption(
   contracts: { id: number }[]
 ) {
   return getGrid(profile, (grid) => {
-    return Promise.all(contracts.map(({ id }) => _getConsumption(id, grid)));
+    return Promise.all(contracts.map(({ id }) => _getConsumption(+id, grid)));
```

Comment:
What if any on of these promises failed? is it the required behavior? I'm thinking of using `allSettled` instead?

Reply:
I already handled the promise failure using `_getConsumption`